src/main/java/org/broadinstitute/hellbender/tools/spark/sv/ContigAligner.java

```diff
@@ -0,0 +1,272 @@
+        for (final Tuple2<RunSGAViaProcessBuilderOnSpark.ContigsCollection.ContigID, RunSGAViaProcessBuilderOnSpark.ContigsCollection.ContigSequence> contig : contigsCollection.getContents()) {
+            if (contig._1.equals(new RunSGAViaProcessBuilderOnSpark.ContigsCollection.ContigID(contigId))) {
+                final String sequence = contig._2.toString();
+                return sequence.substring(startInContig, endInContig);
+            }
+        }
+        throw new GATKException("Could not find contig " + contigId);
+    }
+
+    /**
+     * Wrap a contig sequence in a ShortRead object and pass it to BWA to align
+     * @param bwaMem
+     * @param contigId
+     * @param sequence
+     * @return
+     * @throws IOException
+     */
+    private AlnRgn[] bwaAlignSequence(final BwaMem bwaMem, final String contigId, final byte[] sequence) throws IOException {
+        final ShortRead contigShortRead = new ShortRead(contigId, sequence, qualSequence(sequence.length));
+        return bwaMem.align(contigShortRead);
+    }
+
+    /**
+     * Generate a bogus base quality sequence to pass in for the aligned contig (since the jBWA API requires that reads have qualities)
+     * @param length
+     * @return
+     */
+    private byte[] qualSequence(final int length) {
+        final byte[] quals = new byte[length];
+        for (int i = 0; i < quals.length; i++) {
+            quals[i] = 'A';
+        }
+        return quals;
+    }
+
+    public void close() {
+        bwaMem.dispose();
+    }
+
+    /**
+     * Holds information about a split alignment of a contig, which may represent an SV breakpoint. Each AssembledBreakpoint
+     * represents the junction on the contig of two aligned regions. For example, if a contig aligns to three different regions
+     * of the genome (with one primary and two supplementary alignment records), there will be two AssembledBreakpoint
+     * objects created, one to represent each junction between alignment regions:
+     *
+     * Example Contig:
+     * ACTGACTGCACTGACTGCACTGACTGCACTGACTGCACTGACTGCACTGACTGCACTGACTGCACTGACTGCACTGACTGCACTGACTGCACTGACTGCACTGACTG
+     * Alignment regions:
+     * |---------1:100-200------------|
+     *                                |-----------2:100-200------------------|
+     *                                                                       |----------3:100-200-----------------|
+     * Assmbled breakpoints:
+     * 1) links 1:100-200 to 2:100-200
+     * 2) links 2:100-200 to 3:100-200
+     *
+     * Inserted sequence contains portions of the contig that are aligned to neither region, and therefore may be inserted in
+     * the sample. For example, a translocation breakpoint with a microinsertion:
+     *
+     * Contig:
+     * ACTGACTGACTGACTGACTGACTGACTGACTGACTGACTGACTG
+     * Alignment regions:
+     * |-----1:100-200-------|
+     *                         |-----2:100-200-----|
+     * Inserted sequence:
```

Comment:
Should the second alignment region's left vertical bar match the 'C' instead of 'A'?